core/app/assets/js/layout/header.js

```diff
@@ -2,16 +2,22 @@ import HeaderButtons from './header-buttons';
 import Grid from '../ui/grid/grid';
 
 export default function Header( props ) {
+	const TitleTag = React.useMemo( () => props.titleClickHandler ? 'a' : 'span', [ props.titleClickHandler ] ),
+		titleAttrs = {
+			onClick: props.titleClickHandler,
+			href: 'a' === TitleTag ? '#' : null,
+		};
+
 	React.useEffect( () => {
 		document.title = __( 'Elementor', 'elementor' ) + ' | ' + props.title;
 	} );
 
 	return (
 		<Grid container alignItems="center" justify="space-between" className="eps-app__header">
-			<span className="eps-app__logo-title-wrapper">
+			<TitleTag className="eps-app__logo-title-wrapper" { ...titleAttrs }>
 				<i className="eps-app__logo eicon-elementor"/>
 				<h1 className="eps-app__title">{ props.title }</h1>
-			</span>
+			</TitleTag>
```

Comment:
You can attach the onclick to the span.

Reply:
Yeah, just want to ensure that the code is semantically correct.
I will change it if this is not Important.

Reply:
It can always be an `a` that makes the redirect, without the handler as a prop. (override current commit)